layer1/monitor/events/snapshots.go

```diff
@@ -12,8 +12,8 @@ import (
 
 // ProcessSnapshotTaken handles receiving snapshots
 func ProcessSnapshotTaken(eth layer1.Client, contracts layer1.AllSmartContracts, logger *logrus.Entry, log types.Log, adminHandler monInterfaces.AdminHandler, taskRequestChan chan<- tasks.TaskRequest) error {
-
```

Comment:
Unless these changes are needed for this PR, better to pull them out into a separate PR that will get much faster review.

Reply:
Unfortunately, we need this file here, if not sync will break when new nodes join the network. Since we are using the ring buffer now, we don't have a way to get more than 6 snapshots in the past by querying the contract as we used to. Now, the snapshots data (Bclaims) is added to the event and digested with the rest of the events.